import/controls/permissionmanager.h

```diff
@@ -0,0 +1,30 @@
+#ifndef PERMISSIONMANAGER_H
+#define PERMISSIONMANAGER_H
+
+#include <QObject>
+
+class PermissionManager : public QObject
+{
+    Q_OBJECT
+public:
+    PermissionManager(QObject *parent = nullptr);
+
+    // permission cases: 'unknown', 'allow', 'deny', 'prompt'
+    Q_INVOKABLE void add(const QString &uri, const QString &type, const QString &permission);
+    Q_INVOKABLE void remove(const QString &uri, const QString &type);
+    Q_INVOKABLE void removeAll();
+    Q_INVOKABLE void requestUriPermissions(const QString &uri);
+
+    static void sendRequest(const QString &message
+                       , const QString &uri = ""
+                       , const QString &type = ""
+                       , const QString &permission = "");
+
+signals:
+    void finished(const QVariant &data);
```

Comment:
You should leave a comment that it's the result of `requestUriPermissions()` call.